simperby/src/node/state_storage.rs

```diff
@@ -16,13 +12,107 @@ pub enum StateStorageError {
     StorageIntegrityError(String),
 }
 
+type Error = StateStorageError;
+
+impl From<simperby_kv_storage::Error> for Error {
+    fn from(e: simperby_kv_storage::Error) -> Self {
+        Error::StorageError(e)
+    }
+}
+
 impl From<Error> for SimperbyError {
     fn from(e: Error) -> Self {
-        unimplemented!()
+        match e {
+            Error::InvalidOperation(e) => SimperbyError::InvalidOperation(e),
+            Error::StorageError(e) => SimperbyError::StorageError(e),
+            Error::StorageIntegrityError(e) => SimperbyError::StorageIntegrityError(e),
+        }
     }
 }
 
-type Error = StateStorageError;
+/// Reads an essential (existence-guaranteed) item from the storage.
+async fn get_json_from_storage<T>(storage: &dyn KVStorage, key: Hash256) -> Result<T, Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    let value = match storage.get(key.clone()).await {
+        Ok(x) => x,
+        Err(simperby_kv_storage::Error::NotFound) => {
+            return Err(Error::StorageIntegrityError(format!(
+                "missing essential item in the state stroage: {}",
+                key
+            )))
+        }
+        Err(e) => return Err(e.into()),
+    };
+    serde_json::from_slice(&value).map_err(|e| Error::StorageIntegrityError(e.to_string()))
+}
+
+async fn get_hash_from_storage(storage: &dyn KVStorage, key: Hash256) -> Result<Hash256, Error> {
+    let value = match storage.get(key.clone()).await {
+        Ok(x) => x,
+        Err(simperby_kv_storage::Error::NotFound) => {
+            return Err(Error::StorageIntegrityError(format!(
+                "missing essential item in the state stroage: {}",
+                key
+            )))
+        }
+        Err(e) => return Err(e.into()),
+    };
+    let msg = if value.len() > 1024 {
+        format!(
+            "item hash is corrupted: (too long; emited; {})",
```

Comment:
```suggestion
            "item hash is corrupted: (too long; emitted; {})",
```

Reply:
I guess I gotta test myself for brain diseases